### DISCUSSION TITLE
Need help configuring

### DISCUSSION THREAD

**Discussion by vasfvitor** [Vagr9K/gatsby-advanced-starter#165] [Q&A]
I've been playing everyday with this starter about a week or more, shadowing is working fine (more or less) Netlify CMS is ok. But I'm still struggling to config `gatsby-config.yml` and some basic stuff (I guess it's basic)

From what I've seen, I don't need to configure any plugin because they are all pre-configured in `gatsby-theme-advanced` right? I only need to configure if I install a new plugin?

Another problem I'm having is about the part to config the theme options https://github.com/Vagr9K/gatsby-advanced-starter/blob/master/themes/amaranth/README.md#configuration 
The general website metadata is working, the logo and favicon too but PWA icons it's not.

Even though it's working I'm not sure I did the right setup. 

Also I don't know if I'm supposed to put these options in the `gatsby-config` file inside the plugin `options` or outside the `options` or even in another file. Or doesn't matter?

About this part, what is it for?
`const config: SiteConfig = {`

My `gatsby-config.yml` file starts like that, I removed that `const config: SiteConfig = {`
```
module.exports = {
  plugins: [{
      resolve: `gatsby-theme-amaranth`,
      options: {
        basePath: "/",
        website: {
```
Earlier, I was looking at  examples/amaranth-demo/gatsby-config.js  and there's a list of icons, I copied the whole file and replaced with my info and the icons in /docs/icons with mine, but still I'm not sure I did it right, I think I didn't. Also, is it necessary to do it?

Actually I only got it working because I just copied everything from this file: https://github.com/Vagr9K/gatsby-advanced-starter/blob/master/examples/amaranth-demo/gatsby-config.js

But still, I don't know if it's necessary because the theme already has a `gatsby-config` file...
The LINKS in the footer are also gone, I'm still troubleshooting it, it may be from the shadowing, but may be not. Another thing that's missing is the Author section.
 
Well, I'm really confused as you can see, thanks.

**Comment by vasfvitor** [comment]
For future reference, based on Gatsby Docs:
Shadowed files goes in `src` project root folder:
`src/theme-name`
ex: `src/gatsby-theme-amaranth/`

While the original is
from nodes_modules `/gatsby-theme-amarant/scr/`
`themes/amaranth/src` 

Ex:
Shadow:
`src/gatsby-theme-amaranth/components/Article/Share/index.tsx`

Original:
`themes/amaranth/src/components/Article/index.tsx`

I'm not sure about best practices, but sometimes I copy both the file I'm shadowing and the imports, 
but sometimes I only copy the file I'm gonna shadow and change the `import path`.

**Comment by vasfvitor** [comment]
Localization of dates.

As the theme use[ date-fns](https://date-fns.org/) module you can import the localization for your language and translate the months in posts. I found the locale options in `node_modules/date-fns/locale`

For this example the shadow will be in this folder:

`src\gatsby-theme-amaranth\components\ArticleInfo\index.tsx`

This file show the published date of the post in the blog.

There you `import` your locale.
Mine is pt-BR:
added this line `import { ptBR } from "date-fns/locale";` after/below `import { format } from "date-fns";`

This section is the section to edit:
`ArticleInfo\index.tsx`
```
  const publicationDate = `${categoryUrl ? "\u00A0⋅ " : ""}${format(
    post.datePublished,
    "LLL d, y"  
  )}`;

```
 `LLL d, y` this is the date format from date-fns, [here ](https://date-fns.org/v2.24.0/docs/format) is the full documentation.
Basically, patterns with `L `is month, `d` is day and `y` year. I needed it to be day/month/year so I changed to `d LLL, y`

After that I added the option (I don't know how it's called)  `{ locale: ptBR }`
```
  const publicationDate = `${categoryUrl ? "\u00A0⋅ " : ""}${format(
    post.datePublished,
    "d LLL, y", { locale: ptBR }
  )}`;

```

In the same file there's the "time to read" estimation, but it's just a string that you can change manually.

**Comment by Vagr9K** [comment]
> But still, I don't know if it's necessary because the theme already has a gatsby-config file...

You need to set these options since theme's `gatsby-config` has nothing to do with your's. It just sets the baseline with no defaults.

> The LINKS in the footer are also gone, I'm still troubleshooting it, it may be from the shadowing, but may be not. Another thing that's missing is the Author section.

Not setting the `user` information properly in `gatsby-config` would cause that.

> From what I've seen, I don't need to configure any plugin because they are all pre-configured in gatsby-theme-advanced right? I only need to configure if I install a new plugin?

Mostly, unless you'd like to add another plugin or override existing configurations.

> About this part, what is it for?
> const config: SiteConfig = {

That's a TypeScript type. I'm not sure where you got that snippet from though, since `gatsby-config` follows a different format.

**Reply by vasfvitor** [reply]
> That's a TypeScript type. I'm not sure where you got that snippet from though
I got it from this README, in "Configuration - Available options are:" https://github.com/Vagr9K/gatsby-advanced-starter/blob/master/themes/amaranth/README.md#configuration

Indeed, LINKS is working now, it's fine!

Thanks again for your help.
